graph-proxy/src/graphql/workflows.rs

```diff
@@ -270,6 +313,78 @@ impl Task {
     }
 }
 
+#[allow(clippy::missing_docs_in_private_items)]
+#[derive(Debug)]
+enum Tasks {
+    Fetched(Vec<Task>),
+    UnFetched(Arc<Metadata>),
+}
+#[Object]
+impl Tasks {
+    #[allow(clippy::missing_docs_in_private_items)]
+    async fn tasks(&self, ctx: &Context<'_>) -> anyhow::Result<Vec<Task>> {
+        match self {
+            Tasks::Fetched(tasks) => Ok(tasks.clone()),
+            Tasks::UnFetched(metadata) => {
+                let server_url = ctx.data_unchecked::<ArgoServerUrl>().deref();
+                let auth_token = ctx.data_unchecked::<Option<Authorization<Bearer>>>();
+                let mut url = server_url.clone();
+                let namespace = format!(
+                    "{}{}-{}",
+                    metadata.proposal_code, metadata.proposal_number, metadata.visit
+                );
+                url.path_segments_mut().unwrap().extend([
+                    "api",
+                    "v1",
+                    "workflows",
+                    &namespace,
+                    &metadata.name,
+                ]);
+                let request = if let Some(auth_token) = auth_token {
+                    CLIENT.get(url).bearer_auth(auth_token.token())
+                } else {
+                    CLIENT.get(url)
+                };
+                let nodes = request
+                    .send()
+                    .await?
+                    .json::<APIResult<argo_workflows_openapi::IoArgoprojWorkflowV1alpha1Workflow>>()
+                    .await?
+                    .into_result()?
+                    .status
+                    .unwrap() //  Safe as the status field is always present
+                    .nodes;
+                Ok(match Tasks::into_tasks(TaskMap(nodes), metadata.clone())? {
+                    Tasks::Fetched(fetched_tasks) => fetched_tasks,
+                    Tasks::UnFetched(_) => vec![],
+                })
+            }
+        }
+    }
+}
+
+impl Tasks {
+    /// Converts [`TaskMap`] into [`Tasks`]`
+    fn into_tasks(
+        taskmap: TaskMap,
+        metadata: Arc<Metadata>,
+    ) -> Result<Tasks, WorkflowParsingError> {
```

Comment:
Surely this should be implemented on `TaskMap` using `self` as the first parameter? Or be called `from_taskmap`

Reply:
"Would probably make more sense to implement this on `Tasks`" this was your previous comment when I did that 😢 😩

Reply:
Ah yeah, didn't see it took `metadata` too. Definitely have it on `Tasks` and call it `from_taskmap` or something similar

Reply:
But `Tasks::into_tasks` should have the signature `fn into_tasks(self: Tasks) -> Tasks`, which would be a bit silly